test/benchmarks/perftest.pl

```diff
@@ -491,14 +492,16 @@ ()
     return %result;
 }
 
-sub runtest($$)
+sub runtest
 {
     my $testcasename = shift;
     my $variant = shift;
 
     my %results = ("time", -1, "score", -1);
 
-    $other_switches .= " " . @testlistSwitches{$testcasename};
+    #TODO investigate what was meant to go here, other than adding whitespace on the end
+    #This is a hash slice syntax on an array which is empty, so makes no sense at all
+    #$other_switches .= " " . @testlistSwitches{$testcasename};
```

Comment:
Yeah it's weird, the @testlistSwitches is never initialised or changed from being an empty () from what I could see. $other_switches gets lots of switches set elsewhere though. the .= " " on this line would mean a single space is added to the $other_switches string which could be important. But it will never add any extra switches. In general @var{} is a slice of a hash called %var, $var[$index] is accessing an element of an array called @var. This is probably the *worst* bit of Perl syntax IMHO since its always confusing. @testlistSwitches is an array not a hash, so hash slicing on the array will evaluate to undef even with a numeric key (I think).